packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -6,18 +6,31 @@ import { FormResourceService } from '../openmrs-api/form-resource.service';
 import { FormSchemaCompiler } from '@ampath-kenya/ngx-formentry';
 import { LocalStorageService } from '../local-storage/local-storage.service';
 import { FormSchema, Questions } from '../types';
+import * as ngxTranslate from '@ngx-translate/core';
 
 @Injectable()
 export class FormSchemaService {
   constructor(
     private formsResourceService: FormResourceService,
     private localStorage: LocalStorageService,
     private formSchemaCompiler: FormSchemaCompiler,
+    private translate: ngxTranslate.TranslateService,
   ) {}
 
   public getFormSchemaByUuid(formUuid: string, cached: boolean = true): Observable<FormSchema> {
     const cachedCompiledSchema = this.getCachedCompiledSchemaByUuid(formUuid);
+    const currentLang = this.localStorage.getItem('i18nextLng') || 'en';
+    this.translate.setDefaultLang(currentLang);
+
     if (cachedCompiledSchema && cached) {
+      if (cachedCompiledSchema?.translations) {
+        cachedCompiledSchema.translations.forEach((translationResourceUuid: string) => {
```

Comment:
Couldn't we compile the translations as part of this process and skip even reloading them from `localStorage`? That way we avoid issues like "what if local storage is out of date with the schema".